### PR TITLE
BaSP-TG-W08: Employee

### PR DESCRIPTION
**Pull request by kolodani**
end my part of the proyect

### CODE REVIEW

**Comment by leimeter-joaquin** [comment]
Please, re qrite the name of the PR

File: src/index.js [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
+  getEmployeeById,
+  createNewEmployee,
+  editAnEmployee,
+  deleteAnEmployee,
+  fillterAllEmployees,
+} from './resources/employees';
 
 // use "require" to import JSON files
 const admins = require('./data/admins.json');
+// const employees = require('./data/employees.json');
```
Comment (leimeter-joaquin) [inline src/index.js:14]: Remove this line if you are not using employees in this file.

File: src/resources/employees.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    });
+  }
+};
+
+const editAnEmployee = (req, res) => {
+  const id = parseInt(req.params.id, 10);
+  const employee = employees.find((element) => element.id === id);
+  if (employee) {
+    const index = employees.indexOf(employee);
+    const newEmployee = req.body;
```
Comment (leimeter-joaquin) [inline src/resources/employees.js:43]: You should probably not change any field that is not explicitly changed in the query. If I only want to change the email, I shouldn't have to send the name, lastName, phone...

File: src/resources/employees.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    res.status(404).json({ error: 'Employee not found' });
+  }
+};
+
+const deleteAnEmployee = (req, res) => {
+  const id = parseInt(req.params.id, 10);
+  const employee = employees.find((element) => element.id === id);
+  if (employee) {
+    employees.splice(employees.indexOf(employee), 1);
+    res.status(201).json({ data: 'Deleted employee' });
```
Comment (leimeter-joaquin) [inline src/resources/employees.js:61]: This response should be sent in case fs.writeFile doesn't throw an error

File: src/resources/employees.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      }
+    });
+  } else {
+    res.status(404).json({ error: 'Employee not found' });
+  }
+};
+
+const fillterAllEmployees = (req, res) => {
+  let filterAll = employees;
+  if (req.query.id) {
```
Comment (leimeter-joaquin) [inline src/resources/employees.js:74]: You should check that the params sent by the user are correct.

**Review by leimeter-joaquin — CHANGES_REQUESTED** [review]
package-lock should not be changed. Run `npm i`, commit and check again.

File: src/index.js (hunk 1 above)
Comment (kolodani) [reply src/index.js]: line remove

**Review by leimeter-joaquin — APPROVED** [review]

**Pull request merged**